pkg/sbom/core/bom.go

```diff
@@ -238,12 +238,20 @@ func (b *BOM) AddComponent(c *Component) {
 }
 
 func (b *BOM) AddRelationship(parent, child *Component, relationshipType RelationshipType) {
+	// Check the wrong parent to avoid `panic`
+	if parent == nil {
+		return
+	}
 	if parent.id == uuid.Nil {
 		b.AddComponent(parent)
 	}
 
 	if child == nil {
-		b.relationships[parent.id] = nil // Meaning no dependencies
+		// It is possible that `relationships` already contains this parent.
+		// Check this to avoid overwriting.
+		if _, ok := b.relationships[parent.id]; !ok {
+			b.relationships[parent.id] = nil // Meaning no dependencies
+		}
```

Comment:
Is there any problem with overwriting?

Reply:
I probably got the point.

Reply:
I didn't see issues for this case, but when i investigated this case i thought that this is possible.
e.g.
```json
		{
			"relatedSpdxElement": "SPDXRef-Package-2a53baa495b9ddaf",
			"relationshipType": "DEPENDS_ON",
			"spdxElementId": "SPDXRef-Application-150e605f5f17224d"
		},
		{
			"relatedSpdxElement": "SPDXRef-File-123123213123",
			"relationshipType": "DEPENDS_ON",
			"spdxElementId": "SPDXRef-Application-150e605f5f17224d"
		},
```
In this case `packages` doesn't contain `SPDXRef-File-123123213123` -> child will be `nil` -> we will overwrite overwrite 1st relationship